### ISSUE TITLE
H5Fopen error (with fix)

### ISSUE THREAD

**Issue by bnprks** [GreenleafLab/ArchR#703]
I noticed that I'm unable to perform fragment read operations in an ArchR project despite having read permissions on the files in question. This appears to be because the `.h5read` function with `method="fast"` inadvertently requires both read and write permissions.

Example:
```
# Fails if I only have read access to the arrow files
fragment_granges <- getFragmentsFromProject(proj)

# Succeeds even if I only have read access. Note the argument method="slow"
res1 <- ArchR:::.getFragsFromArrow(proj@sampleColData$ArrowFiles[1], chr="chr1", out="GRanges", method="slow")
```

I believe the offending line is https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/ArrowRead.R#L1023
This should be changed to `fid <- H5Fopen(file, "H5F_ACC_RDONLY"))`

**Comment by rcorces** [comment]
This looks true to me. @jgranja24 - I've gone ahead and made the suggested change in the current development branch ("release_1.0.2") via https://github.com/GreenleafLab/ArchR/commit/0291b6fee341a8c910d016a09be432a022fa01d8
If you disagree, feel free to revert. Closing for now.

**Comment by rcorces** [comment]
actually @bnprks - can you test this and close if its ok? sorry - I'm just swamped right now.

**Comment by rcorces** [comment]
@bnprks - bump

**Comment by sunjiahe0502** [comment]
### Error in H5Fopen(file)

Hello, I look forward to your guidance. According to the official tutorial, when I created projHeme3, I reported "Error in H5Fopen(file)"：

The running code is as follows:
`projHeme3 <- addGeneIntegrationMatrix(
    ArchRProj = projHeme2, 
    useMatrix = "GeneScoreMatrix",
    matrixName = "GeneIntegrationMatrix",
    reducedDims = "IterativeLSI",
    seRNA = seRNA,
    addToArrow = TRUE,
    force= TRUE,
    groupList = groupList,
    groupRNA = "BioClassification",
    nameCell = "predictedCell",
    nameGroup = "predictedGroup",
    nameScore = "predictedScore"
`
The  error is as follows：
`2021-09-01 06:45:48 : Transferring Data to ArrowFiles, 5.032 mins elapsed. Error in .safelapply(seq_along(allSamples), function(y) { :  Error Found Iteration 2 :  	[1] "Error in H5Fopen(file) : HDF5. File accessibility. Unable to open file.\n" 	<simpleError in H5Fopen(file): HDF5. File accessibility. Unable to open file.> Error Found Iteration 3 :  	[1] "Error in H5Fopen(file) : HDF5. File accessibility. Unable to open file.\n" 	<simpleError in H5Fopen(file): HDF5. File accessibility. Unable to open file.> In addition: Warning message: In mclapply(..., mc.cores = threads, mc.preschedule = preschedule) :   2 function calls resulted in an error
`

The  log file is as follows：
[ArchR-addGeneIntegrationMatrix-12d46466dba-Date-2021-09-01_Time-06-40-32.log](https://github.com/GreenleafLab/ArchR/files/7088889/ArchR-addGeneIntegrationMatrix-12d46466dba-Date-2021-09-01_Time-06-40-32.log)

**Comment by rcorces** [comment]
@sunjiahe-hub - please dont post the same thing in two different locations. Have you tried this using the `release_1.0.2` branch?
```
devtools::install_github("GreenleafLab/ArchR", ref="release_1.0.2", repos = BiocManager::repositories())
```
Are you by any chance working on a Lustre file system?

**Comment by sunjiahe-hub** [comment]
I’m very sorry, I have a question about the problem classification and submitted two questions in different classification areas. Now I have deleted one of them. I am happy to try the&nbsp;release_1.0.2&nbsp;branch, hoping to solve my problem , I will reply you later.




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "GreenleafLab/ArchR"                                                                                    ***@***.***&gt;;
Date:&nbsp;Wed, Sep 1, 2021 10:36 PM
***@***.***&gt;;
***@***.******@***.***&gt;;
Subject:&nbsp;Re: [GreenleafLab/ArchR] H5Fopen error (with fix) (#703)





 
@sunjiahe-hub - please dont post the same thing in two different locations. Have you tried this using the release_1.0.2 branch?
 devtools::install_github("GreenleafLab/ArchR", ref="release_1.0.2", repos = BiocManager::repositories())  
Are you by any chance working on a Lustre file system?
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by juliabelk** [comment]
@rcorces I had the same problem, loaded the release branch, and it seems to be working now
